### ISSUE TITLE
Definition of "keyboard interface" - add a note about accessibility interface

### ISSUE THREAD

**Issue by TanyaVanWorkum** [w3c/matf#220]

**Comment by JJdeGroot** [comment]
Discussed in yesterday’s meeting.

<details>
<summary>16 July 2025</summary>

[Source](https://www.w3.org/XXXX/YY/ZZ-matf-minutes.html)

~~~text
Accessibility interface proposal

JJ: Please add thoughts on page to github. It's close to having a first definition that could work for the majority of mobile application cases.

<JJ> [w3c/matf#220](https://github.com/w3c/matf/issues/220)

<JJ> sub issue of: [w3c/matf#66](https://github.com/w3c/matf/issues/66)

JJ: Still some discussion whether "keyboard interface" is replaced with "accessibility interface"

<JJ> Proposed definition:

<JJ> accessibility interface

<JJ> interface used by assistive technology software to obtain semantic information about content and interact with mobile applications

<JJ> Note 1

<JJ> An accessibility interface allows assistive technology to access and interpret content even when the mobile application does not contain accessibility support

<JJ> Example

<JJ> A mobile application exposes its content through the platform's accessibility API (such as UIAccessibility on iOS or AccessibilityService on Android). Screen readers, voice control systems, and switch access software can all use this interface to obtain information about the application's content, structure, and state, enabling users to

<JJ> navigate and interact with the application through their preferred assistive technology.

<JJ> Note 2

<JJ> Operation of an application through a keyboard interface does not automatically qualify as operation through an accessibility interface, as keyboard interfaces focus on keystroke input while accessibility interfaces provide broader capabilities for assistive technology software.

JJ: "accessibility interface" can be used more broadly

Joe_Humbert: Do we need to broaden the first definition to "assistive technology software and settings"? There are settings in Android and iOS that aren't considered software.

JJ: Do we focus only on the interaction part or do we also want to emphasis the semantic information that can be obtained?

Joe_Humbert: I do think we should keep the semantic information in there.

ACTION: Attempt to make "assistive technology software" broader to make sure all types of assistive tech are included

rachaely: Is there a definition of assistive technology in WCAG?

<JJ> https://www.w3.org/TR/wcag2ict-22/#dfn-assistive-technologies

<Joe_Humbert> https://www.w3.org/TR/WCAG22/#dfn-assistive-technologies

Joe_Humbert: Maybe we change it from "assistive technology software" to "assistive technology"

<Jamie> prsent+

ACTION: Change 'assistive technology software' (2x) to 'assistive technology' with link to definition

ACTION: Figure out where the accessibility interface definition should be used

JJ: We might need to change the definition of mobile context
~~~
</details>

- ACTION: Attempt to make "assistive technology software" broader to make sure all types of assistive tech are included
- ACTION: Change 'assistive technology software' (2x) to 'assistive technology' with link to definition
- ACTION: Figure out where the accessibility interface definition should be used

**Comment by JJdeGroot** [comment]
Update proposal based on feedback in last week's meeting.

- Changed "assistive technology software" to "assistive technology"

## Accessibility interface

> Interface used by [assistive technology](https://www.w3.org/TR/WCAG22/#dfn-assistive-technologies) to obtain semantic information about content and interact with mobile applications

### Note 1
> An accessibility interface allows [assistive technology](https://www.w3.org/TR/WCAG22/#dfn-assistive-technologies) to access and interpret content even when the mobile application does not contain accessibility support.
>> **Example**
>> A mobile application exposes its content through the platform's accessibility API (such as [UIAccessibility](https://developer.apple.com/documentation/uikit/uiaccessibility-protocol) on iOS or [AccessibilityService](https://developer.android.com/reference/android/accessibilityservice/AccessibilityService) on Android). Screen readers, voice control systems, and switch access software can all use this interface to obtain information about the application's content, structure, and state, enabling users to navigate and interact with the application through their preferred assistive technology.

### Note 2

> Operation of an application through a [keyboard interface](https://www.w3.org/TR/WCAG22/#dfn-keyboard-interface) does not automatically qualify as operation through an accessibility interface, as keyboard interfaces focus on keystroke input while accessibility interfaces provide broader capabilities for assistive technology software.
